packages/headless/src/features/case-assist/case-assist-slice.ts

```diff
@@ -0,0 +1,67 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  getClassifications,
+  getDocumentSuggestions,
+  setCaseAssistId,
+  setCaseInformationValue,
+  setDebug,
+  setUserContextValue,
+} from './case-assist-actions';
+import {getCaseAssistInitialState} from './case-assist-state';
+
+export const caseAssistReducer = createReducer(
```

Comment:
I think that most of these could actually be removed, and you would simple reuse existing controller.

You don't really need to have something special in case assis to "setContext", or to "setDebug", these are already covered by other controllers.

If you absolutely with to expose a `setContext` functionality directly on the case assist controller versus having them simple use the `context` controller, then I would see that as your CaseAssist controller just creating the needed controller internally, and "proxy" their functionality (aka: compose multiple controller together)

Reply:
You have a good point. I did it this way thinking that we might want to pass different context for the search and case assist calls. But, if a customer implementation requires different context values for both, then I guess two distinct engines should be used. Also, having context values in different parts of the state might cause some confusion.

Given your comment, I guess it would make sense to include both the context and debug sections in the case assist, but let the implementer decide whether he needs to use the controllers or not.